packages/odata-common/src/batch-response.ts

```diff
@@ -36,15 +36,15 @@ export interface WriteResponses<DeSerializersT extends DeSerializers>
  */
 export interface ErrorResponse extends BatchResponseTypeGuards<any> {
   /**
-   * TODO-JSDOC.
+   * To make ErrorResponse structurally different and make typeguards work as expected.
```

Comment:
FYI: This is a typical pattern in typescript called `tagged union` or `discriminating unions`. 
https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html#discriminating-unions, so you could write something like: `Tag for identifying the elements of the BatchResponse union type.`